src/main/java/edu/cornell/kfs/tax/service/impl/TaxProcessingV2ServiceImpl.java

```diff
@@ -24,7 +24,8 @@ public class TaxProcessingV2ServiceImpl implements TaxProcessingV2Service {
     private static final Logger LOG = LogManager.getLogger();
 
     private TaxProcessingDao legacyTaxProcessingDao;
-    private TaxFileGenerationService taxFileGenerationService;
+    private TaxFileGenerationService taxFileGenerationServiceFor1042S;
+    private TaxFileGenerationService taxFileGenerationServiceForTransactionPrinting;
     private TaxParameterService taxParameterService;
```

Comment:
Are you re-using a class to do 2 different things? Is the first one generating files?

Reply:
These are two different service implementations with their own distinct tasks. One is responsible for creating the 1042-S Sprintax files, while the other is responsible for printing the transaction rows to a plain text file. The functionals sometimes take the latter file and use it to construct another text file for batch uploads of Transaction Override objects.